### PR TITLE
Update some of the introduction of step-2.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: examples/step-2/doc/intro.dox [hunk 1]
```diff
@@ -77,23 +89,54 @@ number of matrix-vector multiplications to come up with the solution of a
...
+matrix-vector products would take ${\cal O}(N^s)$ operations), 
+but it also requires very specialized solvers such as
 multigrid methods to satisfy the requirement that the solution requires only a
 fixed number of matrix-vector multiplications. We will frequently look at the
 question of what solver to use in the remaining programs of this tutorial.
 
 The sparsity is generated by the fact that finite element shape
-functions are defined locally on individual cells, rather than
+functions are defined <i>locally</i> on individual cells (or, in some cases such
+as the Discontinuous Galerkin metho, on neighboring cells), rather than
```
Comment (drwells) [inline examples/step-2/doc/intro.dox]: method

Comment (drwells) [reply examples/step-2/doc/intro.dox]: I am not sure that I understand this statement about DG methods. Does this state that some functions are defined over multiple cells? That doesn't sound right.

Comment (kronbichler) [reply examples/step-2/doc/intro.dox]: @drwells is right that this does not sound fully correct. I would phrase this as "or, in some cases, such as the discontinuous Galerkin methods, also connect to neighboring cells through face integrals"

Comment (bangerth) [reply examples/step-2/doc/intro.dox]: Yes, all good points.

**Comment by bangerth** [comment]
I've updated the one comment. It became too many parenthetical remarks, so I put this one to the end of the paragraph.

**Pull request merged**